functions/index.js

```diff
@@ -0,0 +1,153 @@
+const axios = require('axios');
+const btoa = require('btoa')
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+
+admin.initializeApp({
+    credential: admin.credential.applicationDefault(),
+    databaseURL: "https://tuneiq.firebaseio.com"
+});
+
+exports.createGame = functions.https.onRequest((request, response) => {
+    const gameGenre = request.query.genre
+    const hostId = request.query.hostId
+    const gameName = request.query.name
+    const password = request.query.password
+    const gameRounds = functions.config().gameconfig.number_of_rounds
+
+    getRoundDetails(gameGenre, gameRounds).then(roundDetails => {
+        gameDetails = {
+            hostId: hostId,
+            name: gameName,
+            password: password,
+            genre: gameGenre,
+            roundDetails: roundDetails,
+            players: [hostId]
+        }
+        admin.firestore().collection("games").add(gameDetails).then(entry => {
+            console.log(entry.id)
+            response.send(entry.id)
+            return entry
+        }).catch(err => {
+            console.log(err)
+            return err
+        })
+        return gameDetails
+    }).catch(err => {
+        console.log(err)
+        return err
+    })
+})
+
+function getRoundDetails(gameGenre, gameRounds) {
+    const optionsPerRound = functions.config().gameconfig.number_of_options
+    return getSpotifyToken().then(res => {
+        const access_token = res
+        return getSpotifyTracks(gameGenre, gameRounds, access_token, optionsPerRound).then(tracks => {
+            var answerTracks = []
+
+            // there is a chance that there will not be enough tracks with previews
+            // TODO: add fallback logic to get more tracks with previews
+            tracks.every(track => {
+                if (track.preview_url !== null) {
+                    answerTracks.push({
+                        title: track.name,
+                        artists: track.artists.map(artist => artist.name),
+                        album: track.album.name,
+                        previewUrl: track.preview_url
+                    })
+                }
+                if (answerTracks.length === gameRounds) {
+                    return false
+                }
+                return true
+            })
+
+            //making round objects
+            gameRoundDetails = []
```

Comment:
shouldn't declare variables like this - this will use global scope (use const here)